File: backend/app/schemas/application.py
import uuid
from datetime import datetime
from typing import Optional, Any

from pydantic import BaseModel, Field, field_serializer

from app.models.application import ApplicationStatus


class ContactInfo(BaseModel):
    phone: Optional[str] = None
    email: Optional[str] = None
    wechat: Optional[str] = None
    other: Optional[str] = None


class WorkExperience(BaseModel):
    company: str
    position: str
    start_date: str
    end_date: Optional[str] = None
    description: str


class ProjectExperience(BaseModel):
    name: str
    role: str
    start_date: str
    end_date: Optional[str] = None
    description: str
    technologies: list[str] = []


class Education(BaseModel):
    school: str
    major: str
    degree: str
    start_date: str
    end_date: Optional[str] = None


class Certificate(BaseModel):
    name: str
    date: Optional[str] = None


class StructuredResume(BaseModel):
    name: str
    work_experience_years: int = 0
    education_level: Optional[str] = None
    contact: ContactInfo = ContactInfo()
    work_experience: list[WorkExperience] = []
    project_experience: list[ProjectExperience] = []
    education: list[Education] = []
    certificates: list[Certificate] = []
    skills: list[str] = []
    self_evaluation: Optional[str] = None


class ApplicationCreateRequest(BaseModel):
    """投递申请请求"""

    job_id: str = Field(..., description="岗位ID")
    resume_text: str = Field(..., min_length=1, description="简历文本内容")
    structured_resume: Optional[StructuredResume] = Field(None, description="结构化简历（可选）")
    cover_letter: Optional[str] = Field(None, description="求职信（可选）")


class ApplicationStatusUpdateRequest(BaseModel):
    """更新申请状态请求"""

    status: ApplicationStatus = Field(..., description="新状态")


class ApplicationResponse(BaseModel):
    """申请响应"""

    id: str
    job_id: str
    applicant_id: str
    applicant_name: Optional[str] = None
    job_title: Optional[str] = None
    resume_text: str
    cover_letter: Optional[str] = None
    structured_resume: Optional[dict[str, Any]] = None
    status: ApplicationStatus
    created_at: datetime

    model_config = {"from_attributes": True}

    @field_serializer("id", "job_id", "applicant_id")
    def serialize_uuid(self, value: uuid.UUID | str, _info: Any) -> str:
        return str(value)


class ApplicationListResponse(BaseModel):
    """申请列表响应"""

    total: int
    page: int
    page_size: int
    items: list[ApplicationResponse]

File: frontend/src/features/applications/types/application.ts
import type { ApplicationStatus } from "@/shared/constants/applicationStatus";

export interface Contact {
  phone?: string;
  email?: string;
  wechat?: string;
  other?: string;
}

export interface WorkExperience {
  company: string;
  position: string;
  start_date: string;
  end_date?: string;
  description: string;
}

export interface ProjectExperience {
  name: string;
  role: string;
  start_date: string;
  end_date?: string;
  description: string;
  technologies: string[];
}

export interface Education {
  school: string;
  major: string;
  degree: string;
  start_date: string;
  end_date?: string;
}

export interface Certificate {
  name: string;
  date?: string;
}

export interface DimensionScore {
  name: string;
  score: number;
  weight: number;
  reason: string;
}

export interface EvaluationDetail {
  application_id: string;
  applicant_name: string;
  ai_score: number;
  ai_evaluation: DimensionScore[];
  ai_decision: "recommend" | "reject" | "neutral";
  ai_decision_reason: string;
}

export interface StructuredResume {
  name: string;
  work_experience_years: number;
  education_level?: string;
  contact: Contact;
  work_experience: WorkExperience[];
  project_experience: ProjectExperience[];
  education: Education[];
  certificates: Certificate[];
  skills: string[];
  self_evaluation?: string;
}

export interface Applicant {
  id: string;
  applicant_name: string;
  resume_text: string;
  cover_letter?: string;
  structured_resume?: StructuredResume;
  status: ApplicationStatus;
  // AI evaluation fields (Phase 2)
  ai_score?: number;
  ai_evaluation?: DimensionScore[];
  ai_decision?: string;
  ai_decision_reason?: string;
  ai_evaluated_at?: string;
}

export interface MyApplication {
  id: string;
  job_id: string;
  job_title: string;
  company_name?: string;
  resume_text: string;
  cover_letter?: string;
  status: ApplicationStatus;
  created_at: string;
}

export interface PaginatedResponse<T> {
  total: number;
  page: number;
  page_size: number;
  items: T[];
}

export interface CreateApplicationPayload {
  job_id?: string;
  resume_text: string;
  structured_resume: StructuredResume;
  cover_letter?: string;
}
